contracts/colony/ColonyFunding.sol

```diff
@@ -95,24 +95,34 @@ contract ColonyFunding is ColonyStorage, PatriciaTreeProofs { // ignore-swc-123
     }
   }
 
+  int256 constant MAX_PAYOUT_MODIFIER = int256(WAD);
+  int256 constant MIN_PAYOUT_MODIFIER = -int256(WAD);
+
   function claimExpenditurePayout(uint256 _id, uint256 _slot, address _token) public
   stoppable
   expenditureExists(_id)
   expenditureFinalized(_id)
   {
     Expenditure storage expenditure = expenditures[_id];
     ExpenditureSlot storage slot = expenditureSlots[_id][_slot];
-    require(add(expenditure.finalizedTimestamp, slot.claimDelay) <= now, "colony-expenditure-cannot-claim");
+
+    require(
+      add(expenditure.finalizedTimestamp, add(expenditure.globalClaimDelay, slot.claimDelay)) <= now,
+      "colony-expenditure-cannot-claim"
+    );
 
     FundingPot storage fundingPot = fundingPots[expenditure.fundingPotId];
     assert(fundingPot.balance[_token] >= fundingPot.payouts[_token]);
 
     uint256 initialPayout = expenditureSlotPayouts[_id][_slot][_token];
-    uint256 payoutScalar = uint256(slot.payoutModifier + int256(WAD));
+    delete expenditureSlotPayouts[_id][_slot][_token];
+
+    int256 payoutModifier = imin(imax(slot.payoutModifier, MIN_PAYOUT_MODIFIER), MAX_PAYOUT_MODIFIER);
```

Comment:
Well, you mean expenditure state change :) 

Yeah, I was looking at `setExpenditurePayoutModifier` and realized that the checks in that function can't be replicated in the general state change function, at least not without a lot of headache. So it seemed like any checks needed to come later, once the values are being used.

I do think this is a special case though since other values aren't constrained in the same ways. What else could expenditure state change mess up? I'm using "mess up" here to specifically mean taking values outside their intended range, not taking values to unexpected places within their intended range.

Reply:
> I'm using "mess up" here to specifically mean taking values outside their intended range, not taking values to unexpected places within their intended range.

With that definition, I don't think there's anything else.